src/magneticField/CMZField.cpp

```diff
@@ -0,0 +1,458 @@
+	}
+	else
+	{
+    	A= 1.;
+	}
+	return -1/eta *r/(r+1.) * BrAz(r,phi,z,m,B1,eta,R,rr) * A;
+}
+
+// azimuthal field in cartesian  coordinates// for molecular cloud region
+double CMZField::BxAz(double x, double y, double z, double m, double B1, double eta, double R, double rr) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    return (cos(phi)-1./eta * sin(phi)) *BrAz(r,phi,z,m,B1,eta,R,rr);
+}
+
+double CMZField::ByAz(double x, double y, double z, double m, double B1, double eta, double R, double rr) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    return (sin(phi)+1./eta*cos(phi))*BrAz(r,phi,z,m,B1,eta,R,rr);
+}
+
+double CMZField::BzAz(double x, double y, double z, double m, double B1, double eta, double R,double rr) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    return 0;//z component of the magnetic field is in molecular clouds zero
+}
+
+// Polodial field in cartesian coordinates// for non-thermal filament and the intercloud region
+double CMZField::ByPol(double x, double y, double z, double B1, double a1, double a2, double eta) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    double B2=B1/eta;// B1 is the observed magnetic field stremgth and eta the normalization factor otained from the integrations due to the normalization
+    double L=scale(a2/2.,1);
+    double a= 1/pow(scale(a1/2.,2),2.);
+    return sin(phi)*Br(r,z,B2,a,L);
+}
+
+double CMZField::BxPol(double x, double y, double z, double B1, double a1, double a2, double eta) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    double B2=B1/eta;
+    double L=scale(a2/2.,1);
+    double a= 1/pow(scale(a1/2.,2),2.);
+    return cos(phi)*Br(r,z,B2,a,L);
+}
+
+double CMZField::BzPol(double x, double y, double z, double B1, double a1, double a2, double eta) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    double B2=B1/eta;
+    double L=scale(a2/2.,1);
+    double a= 1/pow(scale(a1/2.,2),2.);
+    return Bz(r,z,B2,a,L);
+}
+/////end model C + azimuthal component
+
+bool CMZField::getUseMCField() const {
+    return useMCField;
+}
+bool CMZField::getUseICField() const {
+    return useICField;
+}
+bool CMZField::getUseNTFField() const {
+    return useNTFField;
+}
+bool CMZField::getUseRadioArc() const {
+    return useRadioArc;
+}
+
+void CMZField::setUseMCField(bool use) {
+    useMCField = use;
+}
+void CMZField::setUseICField(bool use) {
+    useICField = use;
+}
+void CMZField::setUseNTFField(bool use) {
+    useNTFField = use;
+}
+void CMZField::setUseRaidoArc(bool use) {
+    useRadioArc = use;
+}
+
+// Parameter identification:
+// r: radius in zylindrical coordinates
+// z= z component in zylindircal coordintes
+// phi: phi component in zylindircal coordinates
+// B1: observed magnetic field strength
+// B2: normalization factor according to B1
```

Comment:
I thought eta is the normalization factor?
Is B2 the normalized field strength or a normalization factor?

Reply:
Yes, B2 is the normalized field strength. I have changed the description.